src/web3_extentions/tx_utils.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from typing import Optional
+
+from eth_account.signers.local import LocalAccount
+from web3.exceptions import ContractLogicError
+from web3.module import Module
+from web3.types import TxParams
+
+from src.metrics.prometheus.basic import TX_SEND, TX_FAILURE
+
+
+logger = logging.getLogger(__name__)
+
+
+class TransactionUtils(Module):
+    @staticmethod
+    def check_transaction(transaction, from_address: str) -> bool:
+        """
+        Returns:
+        True - transaction succeed.
+        False - transaction reverted.
+        """
+        logger.info({"msg": "Check transaction locally.", "value": transaction.args})
+
+        try:
+            result = transaction.call({"from": from_address})
+        except ContractLogicError as error:
+            logger.warning({"msg": "Transaction reverted.", "error": str(error)})
+            return False
+
+        logger.info({"msg": "Transaction executed successfully.", "value": result})
+        return True
+
+    def sign_and_send_transaction(
+        self,
+        transaction: TxParams,
+        gas_limit: int,
+        account: Optional[LocalAccount] = None,
+    ):
+        if not account:
+            logger.info({"msg": "No account provided. Dry mode."})
+            return
+
+        pending_block = self.w3.eth.get_block("pending")
+
+        tx = transaction.build_transaction(
+            {
+                "from": account.address,
+                "gas": gas_limit,
+                "maxFeePerGas": pending_block.baseFeePerGas * 2 + self.w3.eth.max_priority_fee,
+                "maxPriorityFeePerGas": self.w3.eth.max_priority_fee,
+                "nonce": self.w3.eth.get_transaction_count(account.address),
+            }
+        )
+
+        signed_tx = self.w3.eth.account.sign_transaction(tx, account._private_key)
+
+        tx_hash = self.w3.eth.send_raw_transaction(signed_tx.rawTransaction)
+        TX_SEND.inc()
```

Comment:
do we have a separate label sets for metrics for each oracle modules? if not may be we need to have them